### ISSUE TITLE
Release r21 fails on older firmware

### ISSUE THREAD

**Issue by prkiso** [TiniVi/safehax#18]
Two reports of PM INIT FAILED on 9.x firmware. Can't give much more information other than r19 worked on both.

**Comment by MisterWil** [comment]
Can confirm I had this same issue on 9.2. udsploit/fasthax seemed to work and I kept getting PM INIT FAILED. Switched to r19 and it worked fine with udsploit.

**Comment by purpleorangegames** [comment]
Same as above for me.

**Comment by LMLB** [comment]
Same here (O3DS 10.3).

**Comment by TiniVi** [comment]
That's interesting... None of the code before PM init has been changed between revisions, but I'll look into it.

**Comment by purpleorangegames** [comment]
I installed a9lh on a N3DS on 11.3 and on a 2DS on 9.2 today. I had to use r19 on 9.2. The only thing I changed from them was the Otherapp payload and the sound file for soundhax. I don't know if this information might help you.

**Comment by TiniVi** [comment]
Can someone try compiling from the source and running on any older firmware? I have a hunch on why it's not working.

**Comment by purpleorangegames** [comment]
I can test it on 9.2, what are the requirements to compile?

**Comment by Kuruyia** [comment]
Just compiled safehax from source, my safehax.3dsx and the one on the release page are different.

[safehax.zip](https://github.com/TiniVi/safehax/files/921147/safehax.zip)

**Comment by TiniVi** [comment]
I think my build environment is just borked since I was fiddling with stuff between 11.3 and 11.4's release. So yeah, if someone could run @Arc13's compiled copy and verify, I'd appreciate it.

**Comment by purpleorangegames** [comment]
I'm testing it right now, it doesn't give me the 'pm init failed' anymore but just stays on a black screen instead.

**Comment by pirate5314** [comment]
Nope doesnt work either, running on 9.6.0.17 r21 gives PM INIT failure. and r19 and Arcs both give failed to read arm9loader.  

edit r18 did the job. it seems this time it did read the safehaxpayload.bin

**Comment by Kuruyia** [comment]
This version should work on 9.x :

<Nope, doesn't work>

**Comment by pirate5314** [comment]
thanks will try

**Comment by Kuruyia** [comment]
Sorry, I forgotten something important, now it should work:

[Safehax3.zip](https://github.com/TiniVi/safehax/files/921247/Safehax3.zip)

**Comment by DiamondStashers** [comment]
Arc the new one doesn't work either its just a black screen

**Comment by Kuruyia** [comment]
And with just the recompiled version ?

https://github.com/TiniVi/safehax/files/921147/safehax.zip

**Comment by DiamondStashers** [comment]
@Arc13 Ye I've tried both versions with only the recompiled mode

**Comment by Kuruyia** [comment]
Strange, it the recompiled version worked with someone that has problem to execute safehax, just to be sure, here's a version with GodMode9 bundled:

[Safehax embedded.zip](https://github.com/TiniVi/safehax/files/921937/Safehax.embedded.zip)

**Comment by purpleorangegames** [comment]
I have a9lh installed on 9.2, would it cause the safehax to stay in a black screen?

Because I used r19 and it worked at the time, but now every version I test just goes to a black screen, even r19.

**Comment by Kuruyia** [comment]
Oh, yes, if you have a9lh it's totally normal, it also stays on a black screen in my hacked 3DSs

**Comment by DiamondStashers** [comment]
where do I place the smdh file? that might be the problem

**Comment by DiamondStashers** [comment]
I'm following the 3ds guide atm

**Comment by Kuruyia** [comment]
In the same directory as safehax.3dsx

**Comment by DiamondStashers** [comment]
thanks ill try and see if it works now

**Comment by DiamondStashers** [comment]
@Arc13 thanks so much it is finally working

**Comment by DiamondStashers** [comment]
@Arc13 I'm in godmode what do I do from here? thanks again

**Comment by Kuruyia** [comment]
It's good, but that means that you had a problem with your safehaxpayload.bin, it's either named "safehaxpayload.bin.bin" (make sure your extensions are shown), or your sd card is maybe corrupted.

**Comment by DiamondStashers** [comment]
@Arc13 it might have been one of them what do I do from godmode it gives options for different drives

**Comment by Kuruyia** [comment]
Nothing, I will give you safehax with the good payload, at this point you should have booted into SafeCTRTransfer and not GodMode9

**Comment by DiamondStashers** [comment]
@Arc13 ok thanks

**Comment by Kuruyia** [comment]
This will boot into SafeCTRTransfer:
[safehax safectr.zip](https://github.com/TiniVi/safehax/files/921957/safehax.safectr.zip)

After that, you will just follow [The Guide](https://3ds.guide/safectrtransfer-(homebrew-launcher)) starting with Section III.

**Comment by DiamondStashers** [comment]
yes its working thanks so much @Arc13

**Comment by jsjazz** [comment]
In case this helps anybody else-
I was trying to install A9LH following the 3ds.guide. Old 3DSXL on 9.5. Got as far as run udsploit, start to exit, run safehax- but kept getting the PM INIT error.  Downloaded https://github.com/TiniVi/safehax/files/921247/Safehax3.zip, extracted, renamed both master files to remove "master", copied both to 3ds folder on sdcard, and ran again.
This time udsploit gave message patch success, and safehax started SafeCTRTransfer.
Thanks Arc13.
Don't know why the 3ds.guide didn't work.

**Comment by Chaz679** [comment]
I've done what everyone says but mine does even say pm init error it just goes black then my screen just turns off.Can anyone help?

**Comment by jsjazz** [comment]
Wish I could help, but I just followed the 3ds.guide and stumbled onto this thread when I got the pm init error. Sorted me out so I've now been hacking my Wii U ;-)
Which guide are you using? What steps have you done so far?

**Comment by fmoo** [comment]
For what it's worth, my old 3ds running 10.3.0-28U had this same issue with r21.  @Arc13's safectr build from this thread did work for me.

@Chaz679 - did you remember to run udsploit first?

**Comment by Chaz679** [comment]
@fmoo yes.If I don't run it first it says pm init error but when I do run it first it just black screens and I have to hold down the power button.Im on 9.2.0-20U should I update to a later firmware and start over?

**Comment by DongioAlto** [comment]
@Arc13 thks for your help. I have a problem still. I dont have the error message anymore, but my 3DS just shuts down. I tried over and over, using different files, but the problem remains. How do I solve it?

**Comment by annson24** [comment]
https://github.com/TiniVi/safehax/files/921247/Safehax3.zip,

so this build is a nightly after r21 right? which means it should fix the init fail error on v9.x and should also work on v11.3, correct?

sorry for bumping this issue, just got on the same boat as everybody else here last night and r19 fixed it. Was just asking this to know if I can use this build now instead of r21 in the future.

Thanks in advance.

**Comment by TiniVi** [comment]
@annson24 it's unknown if it works as no one has given me proper feedback with it ¯\_(ツ)_/¯
In any case, it's not meant to fix the issue, it's meant to pinpoint it (although it's not impossible to consider that it might fix it). If you want to try it, hold B while the app loads and it should provide some screen feedback.

**Comment by zeteginara** [comment]
@Arc13 , I was following this guide:

https://3ds.guide/installing-boot9strap-(homebrew-launcher)

Until r21 failed.  r19, as it suggests, gives me the error that it can't find arm9.

I'm just trying to find a way to get into Luma like the guide says.  I got into Godmode9, but I'm confused on what to do next.

I'm just trying to install boot9strap so I can continue the guide.  I'm really new at this.

Any ideas?

**Comment by TheProgrammerEX** [comment]
@zeteginara It looks like (to me anyways) that it cannot find the arm9 payload. Can you verify that you have the payload of your ad card and that it's named correctly?(I think the payload's name is supposed to be savehaxpayload.bin but I'm not for sure so check the guide to verify)

**Comment by zeteginara** [comment]
@TheProgramerEX , I followed the guide to a tee and have every file that it mentions.  

I did at one point download the arm9 payload and copied it over in accordance to this advice:
https://gbatemp.net/threads/failed-to-read-arm9-payload.456423/

...which caused it to open Arm9 once I ran safehax.  Arm9 clearly isn't Luma, so the guide was useless at that point.  I didn't really see a way to install boot9strap.  

Maybe if I knew what I was doing I'd find some way to finish the hack, but I don't.

I have an older 3DS with a firmware of 9.7 something.

**Comment by TiniVi** [comment]
Try the latest commit (r25).
